kadasrouting/gui/valhallaroutebottombar.py

```diff
@@ -0,0 +1,313 @@
+import logging
+import json
+
+from PyQt5.QtGui import QIcon, QColor
+from PyQt5.QtWidgets import QDesktopWidget
+
+from kadas.kadasgui import (
+    KadasBottomBar,
+    KadasMapCanvasItemManager,
+    KadasLayerSelectionWidget,
+)
+from kadasrouting.gui.locationinputwidget import (
+    LocationInputWidget,
+    WrongLocationException,
+)
+from kadasrouting.core import vehicles
+from kadasrouting.utilities import iconPath, pushWarning, transformToWGS
+
+from qgis.utils import iface
+from qgis.core import (
+    QgsWkbTypes,
+    QgsVectorLayer,
+    QgsProject,
+)
+from qgis.gui import QgsRubberBand, QgsMapToolPan
+
+from kadasrouting.core.optimalroutelayer import OptimalRouteLayer
+from kadasrouting.gui.drawpolygonmaptool import DrawPolygonMapTool
+
+
+AVOID_AREA_COLOR = QColor(255, 0, 0)
+LOG = logging.getLogger(__name__)
+
+
+class ValhallaRouteBottomBar(KadasBottomBar):
+    """
+    This is a meta class and should not be instanciated by itself, as it is has no
+    UI file associated with it. Use this only as Parent class.
+    """
+
+    def __init__(self, canvas, action, plugin):
+        KadasBottomBar.__init__(self, canvas, "orange")
+        self.setupUi(self)
+        self.setStyleSheet("QFrame { background-color: orange; }")
+        self.action = action
+        self.plugin = plugin
+        self.canvas = canvas
+        self.waypoints = []
```

Comment:
should we put the waypoints on the optimalroutebottombar.py, since it's not in the CP dialog?

Reply:
It is _not yet_ in the CP dialog. I'm pretty sure there is the risk that they might want it in the future...